Add OpenShift deployment settings section

asaunier opened the pull request.

asaunier commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
+
+Edit file `.travis.yml` and update the environment variables:
+
+.. prompt:: bash
```
Using another format would make sense but "yaml" or even "text" make sphinx crash on my laptop!?

asaunier commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
+        - OPENSHIFT_PROJECT=gs-gmf-<package name>
+        - HELM_RELEASE_NAMES=int,prod
+        - OPENSHIFT_URL=https://api.openshift-ch-1.camptocamp.com/
+        - DOCKER_USERNAME=<FIXME>
```
OK to publish the actual username here?

asaunier replied:
It's already in the scaffolds: https://github.com/camptocamp/c2cgeoportal/blob/2.4/geoportal/c2cgeoportal_geoportal/scaffolds/create/%2Bdot%2Btravis.yml_tmpl even though only commented out. I guess we can leave it as it.

Do you think it is OK if I actually replace ``<FIXME>`` by the real username C2C uses? Or would it be a security issue?

sbrunner replied:
No, he is already visible in the demo project :-)

asaunier commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
+.. prompt:: bash
+
+    cd <project directory>
+    gopass gs/ci/dockerhub/password
```
Perhaps a doc about gopass would be useful somewhere?

asaunier replied:
OK, but I don't remember what you did and have no link :P

sbrunner replied:
https://github.com/camptocamp/geospatial-ci-pass/blob/master/README.md :-)

asaunier commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
+    travis encrypt DOCKER_PASSWORD=<paste password returned by the line above>
+
+Make sure to add line breaks (escaping with ``\``) when filling the ``secure``
+argument so that line length constraints are fullfilled.
```
Is there a way to do it automatically or should the lines be broken by hand?

asaunier commented:
@jwkaltz said

> If we could make information like "Clone https://github.com/camptocamp/private-geo-charts/ and create file" a bit generic, I think it would make sense to add all this to the official doc.Something like a section "Best practices for OpenShift", we recommend to have a repository named for example "private-geo-charts" which contains blablabla
> (but we do not mention the camptocamp private repository explicitly)

sbrunner commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
+-------------------------------
+
+This section is specific to projects using OpenShift for deployment, relying
+on Travis to build Docker images that are then pushed to Docker Hub.
```
Shouldn't we add a note to informs about that can be done only by Camptocamp people?

sbrunner commented on doc/integrator/docker.rst:
```diff
@@ -80,3 +80,53 @@ You can also remove the unused named images, that should be done manually:
 * Remove the unwanted images with ``docker rmi <image>:<tag>``.
+
+
+Configure project for OpenShift
```
Configure Docker Hub publishing and OpenShift notifications

asaunier commented:
@sbrunner PR updated according to your comments.

I am a bit confused by the error reported by Jenkins (really related to my changes?):
> groovy.lang.MissingMethodException: No signature of method: org.jenkinsci.plugins.workflow.support.steps.build.RunWrapper.getBuildCauses() is applicable for argument types: () values: []

sbrunner commented:
:-/ I just re-run the job :-)

asaunier commented:
> I just re-run the job

Still failing...
But anyway I guess there will be other additions to this doc. I'll see at the end if we still have the problem with Jenkins.

By the way I have made changes according to your last comments: 3162677

asaunier commented:
@sbrunner I have updated the branch and fixed a few spelling errors reported by the checker. Now Jenkins passes.
The docker/OS doc can probably be improved but is it OK for you to merge the PR as is?

sbrunner commented:
+1 :-)
